File: lead-module/src/main/java/org/example/leadmodule/lead/domain/Lead.java
package org.example.leadmodule.lead.domain;

import lombok.Builder;
import lombok.With;
import org.example.activitymodule.ActivityInternalDto;

import java.util.List;
import java.util.UUID;

@Builder
public record Lead(
        UUID id,
        String firstName,
        String lastName,
        String email,
        String phoneNumber,
        String subject,
        String city,
        @With LeadState state,
        @With Boolean isActive,
        UUID owner,
        UUID jobTitle,
        UUID company,
        @With List<ActivityInternalDto> activities
) {
}

File: ux/src/app/models/lead/lead.ts
import {ActivitySummary} from '../activity/activity-summary';

export type Lead = {
  id?: string,
  firstName: string,
  lastName: string,
  email: string,
  phoneNumber?: string,
  subject?: string,
  city?: string,
  state: number,
  isActive?: boolean,
  owner: string,
  jobTitle?: string,
  company?: string,
  activities?: ActivitySummary[]
}
